### PR TITLE
boards/meshme: initial implementation of I2C and SPI peripherals

### PR DESCRIPTION
**Pull request by luisan00**
### Contribution description
This PR add initial support for **I2C** and **SPI** peripherals for **Meshme** board

### Testing procedure
Build and flashing procedures for **Meshme** board should works as expected with the new changes but also with two new test:
- `tests/periph_i2c`   
- `tests/driver_bme280`

This pull request could open a new issue: 
- A low level test for **SPI** peripherals should be added

### Issues/PRs references
Not now but it should be added in the short term

### CODE REVIEW

**Review by eduazocar — COMMENTED** [review]
This looks like good for me, I has some questions and i think this needs do squash to merge it.

File: tests/doc.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
-@defgroup test_group Test
+@defgroup tests Test
```
[the remark this answers is not in the document]
Comment (luisan00) [reply tests/doc.txt:2]: yep, may be better if it's done in a next pull request

File: tests/periph_i2c/Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+include ../Makefile.tests_common
+
+FEATURES_REQUIRED = periph_i2c
+FEATURES_OPTIONAL = periph_i2c_reconfigure
+
+USEMODULE += ztimer
+USEMODULE += ztimer_msec
```
[the remark this answers is not in the document]
Comment (luisan00) [reply tests/periph_i2c/Makefile:7]: good one!

**Comment by luisan00** [comment]
done @eduazocar xtimer deleted

**Comment by luisan00** [comment]
also squashed to 3 commits xD

File: tests/driver_bme280/app.config.test [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# this file enables modules defined in Kconfig. Do not use this file for
+# application configuration. This is only needed during migration.
+CONFIG_MODULE_BMX280=y
+CONFIG_MODULE_BME280_I2C=y
+CONFIG_MODULE_FMT=y
+CONFIG_MODULE_XTIMER=y
```
Comment (eduazocar) [inline tests/driver_bme280/app.config.test:6]: add EOF, or remove de xtimer reference

File: tests/driver_bme280/main.c [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+ * Copyright (C) 2016 Kees Bakker, SODAQ
+ *               2018 Freie Universität Berlin
+ * Copyright (C) 2022 Mesh4all mesh4all.org
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup     test_group_driver_bme280
```
Comment (eduazocar) [inline tests/driver_bme280/main.c:11]:
```suggestion
 * @ingroup     tests_driver_bme280
```

File: tests/driver_bme280/doc.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/**
+* @defgroup      test_group_driver_bme280   Application test for bme280
+* @ingroup       test_group
```
Comment (eduazocar) [inline tests/driver_bme280/doc.txt:3]:
```suggestion
* @defgroup      tests_driver_bme280   Application test for bme280
* @ingroup       tests
```

File: tests/periph_i2c/app.config.test [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+# Enable I2C peripheral driver
+CONFIG_MODULE_PERIPH_I2C=y
+
+# Enable I2C scan
+CONFIG_MODULE_I2C_SCAN=y
+
+# Enable xtimer to wake up periodically
+CONFIG_MODULE_ZTIMER=y
+CONFIG_MODULE_ZTIMER_MSEC=y
```
Comment (eduazocar) [inline tests/periph_i2c/app.config.test:12]: Are these config parameters necessary?

File: tests/periph_i2c/doc.txt [hunk 7]
```diff
@@ -0,0 +1,38 @@
+/**
+* @defgroup      test_group_periph_i2c   Get I2C available ports and scan for devices
+* @ingroup       test_group
```
Comment (eduazocar) [inline tests/periph_i2c/doc.txt:3]:
```suggestion
* @defgroup      tests_periph_i2c   Get I2C available ports and scan for devices
* @ingroup       tests
```

File: tests/periph_i2c/main.c [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @brief       I2C low level test
+ *
+ * @author      Luis A. Ruiz <luisan00@hotmail.com>
+ */
```
Comment (eduazocar) [inline tests/periph_i2c/main.c:21]: This need be included to a doxygen group

File: tests/periph_i2c/main.c [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "periph_conf.h"
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+#include "ztimer.h"
```
Comment (eduazocar) [inline tests/periph_i2c/main.c:31]: I didn't find an use of ztimer in this test, Why does it need it?

**Review by eduazocar — CHANGES_REQUESTED** [review]
I see a good job, but this need a few details to build with the new documentation and to resolve some unused header files.

File: tests/driver_bme280/main.c (hunk 4 above)
Comment (luisan00) [reply tests/driver_bme280/main.c:11]: This was resolved in a past discussion , a new pr should be open and should contain all these changes

File: tests/driver_bme280/doc.txt (hunk 5 above)
Comment (luisan00) [reply tests/driver_bme280/doc.txt:3]: This was resolved in a past discussion , a new pr should be open and should contain all these changes

File: tests/periph_i2c/main.c (hunk 9 above)
Comment (luisan00) [reply tests/periph_i2c/main.c:31]: Yeah, good one

File: tests/periph_i2c/main.c (hunk 8 above)
Comment (luisan00) [reply tests/periph_i2c/main.c:21]: Yep, good catch

File: tests/periph_i2c/doc.txt (hunk 7 above)
Comment (luisan00) [reply tests/periph_i2c/doc.txt:3]: Same here

File: tests/periph_i2c/app.config.test (hunk 6 above)
Comment (luisan00) [reply tests/periph_i2c/app.config.test:12]: Uhmmm, not really, it's only a copy pasta issue xD

**Comment by eduazocar** [comment]
Well I see this already, only need do the squash and that is all

**Review by eduazocar — APPROVED** [review]

**Comment by luisan00** [comment]
> Well I see this already, only need do the squash and that is 

Squash was done just before the comment

**Pull request merged**